test/unit/attributes/attribute_test.rb

```diff
@@ -6,65 +6,80 @@
+    end
   end
 
-  describe 'attribute with a default value set' do
-    it 'returns the user-configured default value if no value is assigned' do
-      attribute = Inspec::Attribute.new('test_attribute', default: 'default_value')
-      attribute.value.must_equal 'default_value'
+  describe 'attribute with a value set' do
+    it 'returns the user-configured value' do
+      attribute = Inspec::Attribute.new('test_attribute', value: 'some_value')
+      attribute.value.must_equal 'some_value'
     end
 
-    it 'returns the user-configured default value if no value is assigned (nil)' do
-      attribute = Inspec::Attribute.new('test_attribute', default: nil)
+    it 'returns the user-configured value if nil is explicitly assigned' do
+      attribute = Inspec::Attribute.new('test_attribute', value: nil)
       attribute.value.must_be_nil
     end
 
-    it 'returns the user-configured default value if no value is assigned (false)' do
-      attribute = Inspec::Attribute.new('test_attribute', default: false)
+    it 'returns the user-configured value if false is explicitly assigned' do
+      attribute = Inspec::Attribute.new('test_attribute', value: false)
       attribute.value.must_equal false
     end
 
-    it 'returns value if overriding the default' do
-      attribute = Inspec::Attribute.new('test_attribute', default: 'default_value')
-      attribute.value = 'test_value'
-      attribute.value.must_equal 'test_value'
+    it 'returns a new value if the value has been assigned by value=' do
+      attribute = Inspec::Attribute.new('test_attribute', value: 'original_value')
+      attribute.value = 'new_value'
+      attribute.value.must_equal 'new_value'
+    end
+
+    it 'accepts the legacy ":default" option' do
+      attribute = Inspec::Attribute.new('test_attribute', default: 'a_default')
+      attribute.value.must_equal 'a_default'
+    end
+
+    it 'accepts the legacy ":default" and ":value" options' do
+      attribute = Inspec::Attribute.new('test_attribute', default: 'a_default', value: 'a_value')
+      attribute.value.must_equal 'a_value'
     end
   end
 
   describe 'validate required method' do
-    it 'does not error if a default is set' do
-      attribute = Inspec::Attribute.new('test_attribute', default: 'default_value', required: true)
-      attribute.value.must_equal 'default_value'
+    it 'does not error if a value is set' do
+      attribute = Inspec::Attribute.new('test_attribute', value: 'some_value', required: true)
+      attribute.value.must_equal 'some_value'
     end
 
-    it 'does not error if a value is specified' do
+    it 'does not error if a value is specified by value=' do
       attribute = Inspec::Attribute.new('test_attribute', required: true)
       attribute.value = 'test_value'
       attribute.value.must_equal 'test_value'
     end
 
     it 'returns an error if no value is set' do
+      # Assigning the cli_command is needed because in check mode, we don't error
```

Comment:
Odd...I feel like we should error...

Reply:
In check mode (`inspec check`) it is assumed that we won't have a --attrs file passed in (we might not even support the option).

So, for better or worse, don't treat a missing-but-required attribute as an error in check mode.  We could change that policy, but remember things like A2 run check.